java/kafka/deployment-oauth.yaml

```diff
@@ -21,10 +21,12 @@ spec:
 apiVersion: kafka.strimzi.io/v1beta1
 kind: KafkaUser
 metadata:
-  name: service-account-java-kafka-producer
+  name: java-kafka-producer
   labels:
     strimzi.io/cluster: my-cluster
 spec:
+  authentication:
+    type: tls
```

Comment:
Why do we use the KafkaUser here? With OAuth, the user would be managed in the OAuth server. So why do we use it?

Same applies to the consumer I guess?

Reply:
Oversight. I forgot to remove it. Removed now.

Reply:
@scholzj I've re-ordered the envVars as recommended. I've also realigned the Producer yaml's so they are the same as Streams and Consumer.

Reply:
Just to be clear ... I would probably drop the KafkaUser completely ... I'm not sure I understand its value here.

Reply:
KafkaUser removed from Producer as it's not being used in Oauth.